File: FureyLib/Misc or Not Mine/Rust/Boilerplates/struct.rs
// .rs
// Name
// Creator

use std::boxed::*;
use std::collections::*;
use std::env::*;
use std::io::*;
use std::mem::*;
use std::ops::*;
use std::rc::*;
use std::result::*;
use std::string::*;
use std::vec::*;

// STRUCTURE VARIABLES

/// Template structure script.
pub struct Struct {
    /// Example x variable.
    pub x: f32,

    /// Example y variable.
    pub y: f32, // New variables here
}

// STRUCTURE METHODS

impl Struct {
    // CONSTRUCTORS

    /// Default constructor.
    pub fn new(x: f32, y: f32) -> Self {
        Struct { x, y }
    }

    // New constructors here

    // METHODS

    /// Returns the magnitude of the given structure.
    pub fn magnitude(&self) -> f32 {
        if self.x == 0.0 && self.y == 0.0 {
            return 0.0;
        }
        (self.x * self.x + self.y * self.y).sqrt()
    }

    /// Normalizes the given structure.
    pub fn normalize(&mut self) -> &mut Self {
        let magnitude: f32 = self.magnitude();
        if magnitude == 0.0 {
            return self;
        }
        self.x /= magnitude;
        self.y /= magnitude;
        self
    }

    // New methods here
}

File: FureyLib/Misc or Not Mine/Haskell/Boilerplates/Struct.hs
-- .hs
-- Name
-- Creator

module Struct where


-- DATA

-- | Template structure script.
data Struct = Struct {
    -- | Example x variable.
    x :: Float,

    -- | Example y variable.
    y :: Float
} deriving (Show, Eq)


-- CONSTRUCTORS

-- | Initializes a new structure.
newStruct :: Float -> Float -> Struct
newStruct x y = Struct { x = x, y = y }

-- | Initializes a zeroed structure.
zeroStruct :: Struct
zeroStruct = Struct { x = 0, y = 0 }


-- FUNCTIONS

-- | Returns the magnitude of the given structure.
structMagnitude :: Struct -> Float
structMagnitude struct
    | x struct == 0 && y struct == 0 = 0
    | otherwise = sqrt (x struct * x struct + y struct * y struct)

-- | Normalizes the given structure.
structNormalize :: Struct -> Struct
structNormalize struct
    | magnitude == 0 = struct
    | otherwise = newStruct (x struct / magnitude) (y struct / magnitude)
    where magnitude = structMagnitude (struct)
